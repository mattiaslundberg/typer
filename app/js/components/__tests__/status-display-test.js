import React from 'react'
import StatusDisplay from '../status-display'
import renderer from 'react-test-renderer'

describe("StatusDisplayTest", () => {
  it('Render not-typed chars', () => {
    const component = renderer.create(
      <StatusDisplay
        typedSequence=""
        fullText={["a", "b"]} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('Render invalid chars', () => {
    const component = renderer.create(
      <StatusDisplay
        typedSequence="cc"
        fullText={["a", "b"]} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('Render valid chars', () => {
    const component = renderer.create(
      <StatusDisplay
        typedSequence="ab"
        fullText={["a", "b"]} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
