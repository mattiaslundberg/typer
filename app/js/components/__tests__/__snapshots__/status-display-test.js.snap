exports[`StatusDisplayTest Render invalid chars 1`] = `
<div>
  <span
    className="invalid-typed">
    a
  </span>
  <span
    className="invalid-typed">
    b
  </span>
</div>
`;

exports[`StatusDisplayTest Render not-typed chars 1`] = `
<div>
  <span
    className="not-typed">
    a
  </span>
  <span
    className="not-typed">
    b
  </span>
</div>
`;

exports[`StatusDisplayTest Render valid chars 1`] = `
<div>
  <span
    className="typed">
    a
  </span>
  <span
    className="typed">
    b
  </span>
</div>
`;
